sensors/rgb/rgb.go

```diff
@@ -0,0 +1,78 @@
+// Package rgb implements the RGB sensor
+package rgb
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"go.viam.com/rdk/components/camera"
+	"go.viam.com/rdk/registry"
+	"go.viam.com/rdk/rimage/transform"
+
+	"go.viam.com/slam/sensors/utils"
+)
+
+// RGB represents an RGB sensor.
+type RGB struct {
+	Name string
+	rgb  camera.Camera
+}
+
+// New creates a new RGB sensor based on the sensor definition and the service config.
+func New(ctx context.Context, deps registry.Dependencies, sensors []string, sensorIndex int) (RGB, error) {
+	name, err := utils.GetName(sensors, sensorIndex)
+	if err != nil {
+		return RGB{}, err
+	}
+
+	newRGB, err := camera.FromDependencies(deps, name)
+	if err != nil {
+		return RGB{}, errors.Wrapf(err, "error getting camera %v for slam service", name)
+	}
+
+	if err = validate(ctx, newRGB); err != nil {
+		return RGB{}, err
+	}
+
+	return RGB{
+		Name: name,
+		rgb:  newRGB,
+	}, nil
+}
+
+// GetData returns data from the RGB sensor. The returned function is a release function
+// that must be called once the caller of GetData is done using the image.
+func (rgb RGB) GetData(ctx context.Context) ([]byte, func(), error) {
+	return utils.GetPNGImage(ctx, rgb.rgb)
+}
+
+func validate(ctx context.Context, rgb camera.Camera) error {
```

Comment:
This code came from [here](https://github.com/viamrobotics/rdk/blob/b4a5473abe9a4721ddfc5300c87927e52c7511fb/services/slam/builtin/builtin.go#L206-L240)